sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedDeltaBinaryPackedReader.java

```diff
@@ -300,8 +300,12 @@ private void unpackMiniBlock() throws IOException {
         bitWidths[currentMiniBlock]);
     for (int j = 0; j < miniBlockSizeInValues; j += 8) {
       ByteBuffer buffer = in.slice(packer.getBitWidth());
-      packer.unpack8Values(buffer.array(),
-        buffer.arrayOffset() + buffer.position(), unpackedValuesBuffer, j);
+      if (buffer.hasArray()) {
```

Comment:
This is definitely right, but do we need to make a check on each `buffer`?

Reply:
Unfortunately, yes. ByteBufferInputStream does not have an API to check if it is backed by direct byte buffers. Let me check if I can have the boolean flag to be resolved once on the initialisation.